web/src/main/java/org/springframework/security/web/authentication/switchuser/SwitchUserFilter.java

```diff
@@ -528,6 +529,15 @@ public void setUsernameParameter(String usernameParameter) {
 	}
 
 	/**
+	 * Allows the role of the switchAuthority to be customized.
+	 * 
+	 * @param switchAuthorityRole the role name. Defaults to {@link #ROLE_PREVIOUS_ADMINISTRATOR}
+	 */
+	public void setSwitchAuthorityRole(String switchAuthorityRole) {
```

Comment:
Can you add an Assert.notNull check here? Please also ensure to add a test for this.